src/librustc_driver/lib.rs

```diff
@@ -1011,6 +1011,10 @@ pub fn handle_options(args: &[String]) -> Option<getopts::Matches> {
         return None;
     }
 
+    if cg_flags.iter().any(|x| *x == "no-stack-check") {
+        println!("warning: the --no-stack-check flag is deprecated and does nothing");
```

Comment:
instead of calling `println!` directly, it would be better to invoke `early_warn`, sort of like [this code](https://github.com/rust-lang/rust/blob/master/src/librustc/session/config.rs#L1612-L1618):

```rust
                    let msg = format!("the option `{}` is unstable and should \
                                       only be used on the nightly compiler, but \
                                       it is currently accepted for backwards \
                                       compatibility; this will soon change, \
                                       see issue #31847 for more details",
                                      opt_name);
                    early_warn(ErrorOutputType::default(), &msg);
```